prov/util/src/util_av.c

```diff
@@ -282,9 +285,14 @@ int ofi_av_insert_addr(struct util_av *av, const void *addr, fi_addr_t *fi_addr)
 			return -FI_ENOMEM;
 		if (fi_addr)
 			*fi_addr = ofi_buf_index(entry);
-		memcpy(entry->addr, addr, av->addrlen);
+
+		// If inserting IPv4 into IPv6 table, clear remainder.
+		memcpy(entry->addr, addr, MIN(addrlen, av->addrlen));
+		if (addrlen < av->addrlen) {
+			memset((char *) entry->addr + addrlen, 0, av->addrlen - addrlen);
+		}
 		ofi_atomic_initialize32(&entry->use_cnt, 1);
-		HASH_ADD(hh, av->hash, addr, av->addrlen, entry);
+		HASH_ADD(hh, av->hash, addr, MIN(addrlen, av->addrlen), entry);
```

Comment:
The changes to this function suggest that additional work is needed in order for the AV to support addresses that are different sizes.  The av->addrlen value is used at least 2 different ways.  One way is as the size of the data being hashed.  A second way is the maximum size of the addresses being inserted into the AV.

I suggest replacing av->addrlen with 2 values based on these use cases.  I.e. add av::maxaddrlen and av::hashaddrlen.  Though, maybe hashaddrlen could instead be the minaddrlen.  This should simplify some of the changes here, such as removing the MIN calls.